ansible/library/fabric_info.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+DOCUMENTATION = '''
+module: fabric_info.py
+Ansible_version_added:  2.0.0.2
+short_description:   Find SONiC Fabric ASIC inforamtion if applicable for the DUT
+Description:
+        When the testbed has Fabric ASICs, this module helps to collect that information
+        which helps in generating the minigraph
+    Input:
+        hwsku num_fabric_asic
+
+    Return Ansible_facts:
+    fabric_info:  SONiC Fabric ASIC information
+
+'''
+
+EXAMPLES = '''
+    - name: get Fabric ASIC info
+      fabric_info: hwsku='ACS-MSN2700' num_fabric_asic=1
+'''
+
+RETURN = '''
+      ansible_facts{
+        fabric_info: [{'asicname': 'ASIC0', 'ip_prefix': '10.1.0.33/32', 'ip6_prefix': 'FC00:1::33/128'},
+                      {'asicname': 'ASIC1', 'ip_prefix': '10.1.0.34/32', 'ip6_prefix': 'FC00:1::34/128'}]
+      }
+'''
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            hwsku=dict(required=True, type='str'),
+            num_fabric_asic=dict(type='str', required=False),
+            asics_host_pfx=dict(type='str', required=False),
+            asics_host_pfx6=dict(type='str', required=False)
+        ),
+        supports_check_mode=True
+    )
+    m_args = module.params
+    try:
+        fabric_info = []
+        # num_fabric_asic may not be present for fixed systems which have no Fabric ASIC.
+        # Then return empty fabric_info
+        if 'num_fabric_asic' not in m_args or int(m_args['num_fabric_asic']) < 1:
+           module.exit_json(ansible_facts={'fabric_info': fabric_info})
+           return
+        num_fabric_asic = int( m_args[ 'num_fabric_asic' ] )
+        asics_host_pfx = str( m_args[ 'asics_host_pfx' ] )
+        asics_host_pfx6 = str( m_args[ 'asics_host_pfx6' ] )
+        for asic_id in range(num_fabric_asic):
+            key = "ASIC%d" % asic_id
+            data = { 'asicname': key,
+                     'ip_prefix': asics_host_pfx % asic_id,
+                     'ip6_prefix': asics_host_pfx6 % asic_id }
```

Comment:
This assumes that variables like `asics_host_pfx` and `asics_host_pfx6` must be a string template. I assume it is like: `10.1.0.%s/32`
In the template, `%s` will be replaced with asic_id which is an integer between `0` and `num_fabric_asic`.

In the example output:
```
      ansible_facts{
        fabric_info: [{'asicname': 'ASIC0', 'ip_prefix': '10.1.0.33/32', 'ip6_prefix': 'FC00:1::33/128'},
                      {'asicname': 'ASIC1', 'ip_prefix': '10.1.0.34/32', 'ip6_prefix': 'FC00:1::34/128'}]
      }
```
It seems that there is no way to generate the IP addresses in the example based on a provided prefix and an integer asic ID.

Reply:
asics_host_pfx is '10.1.0.%d/32'.
do you want more comments in fabric_info ? or add a sample chassis-dut case to testbed-new.yml ?

Reply:
updated to comments with sample arguments that can be passed for asics_host_pfx/asics_host_pfxv6

Reply:
I think there are two issues here:
1. The sample in comments does not match the actual behavior of test code.
2. Does generating ip_prefix for each ASIC based on asic_id and string template matches the behavior of how the actual ASIC ip_prefix is generated? Is the actual ASIC ip_prefix generated from a base IP plus asic_id?

Reply:
Regarding 2., yes the generated ip_prefix, asicname with asic_id does work well in the testings so far.

Reply:
Can you fix issue 2?

Probably the better way is to add asic_id to a base IP to generate ASIC IP?